lib/connectors/memory.js

```diff
@@ -252,6 +252,18 @@ Memory.prototype.save = function save(model, data, options, callback) {
   });
 };
 
+Memory.prototype.replace = function save(model, data, options, callback) {
+  var self = this;
+  var id = this.getIdValue(model, data);
+  var cachedModels = this.collection(model);
+  var modelData = cachedModels && this.collection(model)[id];
+  modelData = modelData && deserialize(modelData);
```

Comment:
AFAICT, `modelData` is used only to compute `isNewInstance`, in which case I think `deserialize(modelData)` is premature pessimization.